extensions/wikia/Flags/specials/scripts/FlagEditForm.js

```diff
@@ -0,0 +1,266 @@
+
+			if (formResources == null || formResources.resources == null) {
+				formResources = loader({
+					type: loader.MULTI,
+					resources: {
+						messages: 'FlagsCreateForm',
+						mustache: '/extensions/wikia/Flags/specials/templates/SpecialFlags_createFlagForm.mustache,/extensions/wikia/Flags/specials/templates/createFormParam.mustache',
+						styles: '/extensions/wikia/Flags/specials/styles/CreateForm.scss'
+					}
+				});
+
+				cache.set(getResourcesCacheKey(), formResources, cache.CACHE_LONG);
+			}
+
+			return formResources;
+		}
+
+		function setupForm(formResources) {
+			loader.processStyle(formResources.styles);
+			mw.messages.set(formResources.messages);
+
+			formData = {
+				/* TODO - Do something with the damn messages */
+				messages: formResources.messages,
+				template: formResources.mustache[0],
+				partialParam: formResources.mustache[1]
+			};
+
+			modalConfig = {
+				vars: {
+					id: 'FlagEditForm',
+					classes: ['flag-edit-form'],
+					size: 'medium', // size of the modal
+					buttons: getButtons(),
+					content: '' // content
+				}
+			};
+		}
+
+		function getButtons() {
+			return [
+				{
+					vars: {
+						classes: ['normal', 'primary'],
+						data: [
+							{
+								key: 'event',
+								value: 'done'
+							}
+						]
+					}
+				},
+				{
+					vars: {
+						data: [
+							{
+								key: 'event',
+								value: 'close'
+							}
+						]
+					}
+				}
+			];
+		}
+
+		function displayFormCreate() {
+			/* TODO - We can get a half-rendered template to avoid escaping messages in front-end */
+			var content = cache.get(getEmptyFormCacheKey());
+			if (content == null) {
+				var formParams = {
+					messages: formData.messages,
+					values: getDropdownOptions({})
+				};
+				content = mustache.to_html(formData.template, formParams, formData.partialParam);
+
+				cache.set(getEmptyFormCacheKey(), content, cache.CACHE_LONG);
+			}
+
+			modalConfig.vars.content = content;
+
+			modalConfig.vars.title = mw.message('flags-special-create-form-title-new').escaped();
+			modalConfig.vars.buttons[0].vars.value = mw.message('flags-special-create-form-save').escaped();
+			modalConfig.vars.buttons[1].vars.value = mw.message('flags-special-create-form-cancel').escaped();
+
+			displayModal();
+		}
+
+		function displayFormEdit(content) {
+			/* TODO - Finish this method */
+			modalConfig.vars.content = content;
+			modalConfig.vars.title = mw.message('flags-special-create-form-title-edit').escaped();
+
+			displayModal();
+		}
+
+		function displayModal() {
+			require(['wikia.ui.factory'], function (uiFactory) {
+				/* Initialize the modal component */
+				uiFactory.init(['modal']).then(function(uiModal){
+					uiModal.createComponent(modalConfig, processInstance);
+				});
+			});
+		}
+
+		function processInstance(modalInstance) {
+			modalInstance.show();
+			modalInstance.bind('done', saveCreateFlagForm);
+			$('.flags-special-form-params-new').on('click', addNewParameterInput);
```

Comment:
I would advice to keep the scope 
$('#FlagEditForm .flags-special-form-params-new')
so we won't be surprized by duplicated actions in the future.